test/unittest/dds/subscriber/DataReaderTests.cpp

```diff
@@ -326,7 +352,73 @@ class DataReaderTests : public ::testing::Test
             reset_lengths_if_ok(code, data_values, infos);
 
             check_instance_methods(instance_ok_code, instance_bad_code, expected_return_loan_ret,
-                    data_reader, data_values, infos);
+                    data_reader, data_values, infos, LENGTH_UNLIMITED, two_instances);
+        }
+    }
+
+    /*
+     * This test checks all variants of read / take in several situations.
+     *
+     * The APIs tested are:
+     * - read_next_sample / take_next_sample
+     * - read / take
+     * - read_next_instance / take_next_instance
+     * - read_instance / take_instance
+     * - return_loan
+     *
+     * The test checks that:
+     * - Calling the APIs on a disabled reader return NOT_ENABLED
+     * - Calling the APIs on an enabled reader with no data return NO_DATA
+     * - Calling the xxx_instance APIs with a wrong instance handle return BAD_PARAMETER
```

Comment:
This part of the test is not implemented.

Reply:
Yes it is. Look for `instance_bad_code` and you'll see it.